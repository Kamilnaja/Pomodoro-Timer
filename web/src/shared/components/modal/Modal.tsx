import React, { MouseEventHandler } from "react";
import ReactModal from "react-modal";
import LoginContainer from "../../../login/container/LoginContainer";
import Settings from "../../../settings/Settings";
import StatisticsContainer from "../../../statistics/container/StatisticsContainer";
import { Modal as ModalEnum } from "../../store/enums/modal.enum";
import "./modal.scss";

export interface ModalProps {
  modalType: ModalEnum;
  handleCloseModal: MouseEventHandler<HTMLButtonElement>;
}

export const Modal = (props: ModalProps) => (
  <ReactModal
    isOpen={props.modalType !== ModalEnum.NULL}
    ariaHideApp={false}
    shouldFocusAfterRender={true}
    shouldCloseOnOverlayClick={true}
    onRequestClose={props.handleCloseModal}
  >
    <div className="modal">
      <header className="modal__header">
        <button
          className="modal__button--close"
          onClick={props.handleCloseModal}
        >
          âœ•
        </button>
      </header>
      {props.modalType === ModalEnum.LOGIN && <LoginContainer></LoginContainer>}
      {props.modalType === ModalEnum.STATS && (
        <StatisticsContainer></StatisticsContainer>
      )}
      {props.modalType === ModalEnum.SETTINGS && <Settings></Settings>}
    </div>
  </ReactModal>
);
