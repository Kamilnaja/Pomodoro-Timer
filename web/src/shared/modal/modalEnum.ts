export enum ModalType {
  LOGIN,
  REGISTER,
  LOGOUT,
  STATS,
  NULL,
}
