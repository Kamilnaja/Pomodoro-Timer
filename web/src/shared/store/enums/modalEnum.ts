export enum Modal {
  LOGIN,
  REGISTER,
  SETTINGS,
  LOGOUT,
  NULL,
}
