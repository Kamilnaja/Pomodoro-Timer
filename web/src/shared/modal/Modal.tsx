import React from 'react';
import ReactModal from 'react-modal';
import { ConnectedLoginContainer } from '../../auth/login/container/LoginContainer';
import { ConnectedLogoutContainer } from '../../auth/logout/container/LogoutContainer';
import { ConnectedRegisterContainer } from '../../auth/register/container/RegisterContainer';
import './modal.scss';
import { ModalType as ModalEnum, ModalType } from './modalEnum';
import { CreatedModal } from './ModalInterface';
import { ModalProps } from './ModalProps';

export const Modal = (props: ModalProps) => {
  const createModalBodyFactory = (modalType: ModalType): CreatedModal => {
    switch (modalType) {
      case ModalEnum.LOGIN:
        return { component: <ConnectedLoginContainer />, title: 'Login' };
      case ModalEnum.REGISTER:
        return { component: <ConnectedRegisterContainer />, title: 'Register' };
      case ModalEnum.LOGOUT:
        return { component: <ConnectedLogoutContainer closeModal={props.closeModal} />, title: 'Logout' };
    }
  };

  const modal: CreatedModal = createModalBodyFactory(props.modalType);
  return (
    <>
      {props.modalType !== ModalEnum.NULL && (
        <ReactModal
          isOpen={true}
          ariaHideApp={false}
          shouldFocusAfterRender={true}
          shouldCloseOnOverlayClick={true}
          onRequestClose={props.closeModal}
          className="modal__content"
          overlayClassName="modal"
        >
          <div className="modal__wrapper">
            <header className="modal__header">
              <h2 className="modal__title">{modal.title}</h2>
              <button className="modal__button--close" onClick={props.closeModal}>
                âœ•
              </button>
            </header>
            {modal.component}
          </div>
        </ReactModal>
      )}
    </>
  );
};
