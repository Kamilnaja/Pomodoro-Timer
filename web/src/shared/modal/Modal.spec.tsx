import { shallow, ShallowWrapper } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import { Modal } from './Modal';
import { ModalType } from './modalEnum';

describe('Modal', () => {
  let wrapper: ShallowWrapper;
  const closeModal = jest.fn();

  beforeEach(() => {
    wrapper = shallow(<Modal modalType={ModalType.LOGIN} closeModal={closeModal} />);
  });

  it('should work', () => {
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
