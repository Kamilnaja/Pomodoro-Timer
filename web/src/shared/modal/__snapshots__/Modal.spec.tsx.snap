// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal should work 1`] = `
<Fragment>
  <Modal
    ariaHideApp={false}
    bodyOpenClassName="ReactModal__Body--open"
    className="modal__content"
    closeTimeoutMS={0}
    contentElement={[Function]}
    isOpen={true}
    onRequestClose={[MockFunction]}
    overlayClassName="modal"
    overlayElement={[Function]}
    parentSelector={[Function]}
    portalClassName="ReactModalPortal"
    preventScroll={false}
    role="dialog"
    shouldCloseOnEsc={true}
    shouldCloseOnOverlayClick={true}
    shouldFocusAfterRender={true}
    shouldReturnFocusAfterClose={true}
  >
    <div
      className="modal__wrapper"
    >
      <header
        className="modal__header"
      >
        <h2
          className="modal__title"
        >
          Login
        </h2>
        <button
          className="modal__button--close"
          onClick={[MockFunction]}
        >
          âœ•
        </button>
      </header>
      <Connect(LoginContainer) />
    </div>
  </Modal>
</Fragment>
`;
